### ISSUE TITLE
Depth Rendering

### ISSUE THREAD

**Issue by karthyyy** [dtorre38/mujoco_opencv#1]
Hi,

  Thanks for integrating mujoco with opencv and sharing the same. I am facing one issue with respect to depth rendering, Earlier, I was using mujoco-viewer library to capture the depth frame. Now I am trying to extract the same with your code. 

I am calling

`depth, offscreen_viewport = get_frame(model, data, opt, scene, context, 'chaser_cam', loc_x=viewport_width - frame_width, loc_y=viewport_height - frame_height, width=frame_width, height=frame_height)  `  

in my simulation loop. The `get_frame` is modified as follows.

```
def get_frame(model, data, opt, scene, context, camera_name, loc_x, loc_y, width=640, height=480):

    # 1. Create a rectangular viewport in the upper right corner for example.
   offscreen_viewport = mj.MjrRect(int(loc_x), int(loc_y), width, height)    
    
    # 2. Specify a different camera view by updating the scene with mjv_updateScene.
    # Set the camera to the specified view
    camera_id = mj.mj_name2id(model, mj.mjtObj.mjOBJ_CAMERA, camera_name)
    offscreen_cam = mj.MjvCamera()
    offscreen_cam.type = mj.mjtCamera.mjCAMERA_FIXED
    offscreen_cam.fixedcamid = camera_id

    mj.mjv_updateScene(model, data, opt, None, offscreen_cam, mj.mjtCatBit.mjCAT_ALL.value, scene)
    
    # 3.Render the scene in the offscreen buffer with mjr_render.
    depth = np.zeros((height*width*3, 1), dtype=np.uint8)  # Placeholder for depth data
    mj.mjr_render(offscreen_viewport, scene, context)

    # 4. Read the depth with mjr_readPixels.
    mj.mjr_readPixels(None, depth, offscreen_viewport, context)
        
    # 5. Depth processing 
    # Shift nearest values to the origin.
    depth-= depth.min()
    # Scale by 2 mean distances of near rays.
    depth= depth/ 2*depth[depth<= 1].mean()
    # Scale to [0, 255]
    depth= 255*np.clip(depth, 0, 1)
          
    # 6. Call mjr_drawPixels using the rectangular viewport you created in step 1.
    # mj.mjr_drawPixels(depth, None, offscreen_viewport, context)
    
    return frame, offscreen_viewport

```

   But depth is not getting rendered. Can you please help with this ?

Thanks and Regards
  Karthik R

**Comment by dtorre38** [comment]
Hello,

The issue is that get_frame() has to be used in conjunction with detect_and_draw_bound(), refer to opencv_tracking.py for implementation on this. However, if you only want to render the image without any further processing then make a call to render_insetwindow().

**Comment by karthyyy** [comment]
Hi,

  Thanks for the reply. I think there is no cross dependency between get_frame() function and detect_and_draw_bound() function. detect_and_draw_bound() is simply a bounding box inserting function based on the input image captured through get_frame(). Please correct if wrong.
 
 My requirement is to extract both RGB and depth image and insert them in the inset, I am facing issues in depth extraction as mentioned in the original query.

Regards
  Karthik R

**Comment by dtorre38** [comment]
As I said before, if you only wish to extract the RGB image and visualize in the inset, utilize render_insetwindow(). This is the course of action that best suits you.

get_frame() obtains the frame RGB pixels and nothing more. That function must be used in conjunction with detect_and_draw_bound() to get an inset window or by passing the RGB pixels to the mujoco function mjr_drawPixels().

**Comment by karthyyy** [comment]
Hi,

  It didnt help. Solved it through direct functions of mujoco rather than custom functions.

Regards
  Karthik

**Comment by dtorre38** [comment]
Hi Karthik,

You're probably no longer in need of this fix, but thanks to #2, the issue you faced is fixed.

I have pushed this change to the code.

Best,
Daniel
